### DISCUSSION TITLE
using single cell model for battery pack

### DISCUSSION THREAD

**Discussion by Jadhav-Sourabh** [pybamm-team/PyBaMM#2224] [General]
is this a good way to use single cell pybamm model for battery pack by creating multiple same variables of SPMe model?
for series connection and parallel connection, I will do simple maths for current and voltage.
for terminal resistance, I will use ohm's law to calculate actual current to pass for the batteries.
for bus-bar resistance and other resistances, I will use another basic electrical formulas.

in liionpack, there is no function to experiment CCCV and other charging curves.
That's why, I used MATLAB/Simulink for battery pack but it is taking very much long time and need to design battery charger seperately. it doesn't have physical model aspect like DFN, SPM.

Please let me know, the above pybamm model method is good way or not?

**Comment by brosaplanella** [comment]
If you want to describe the various cells with the same model and just divide/multiply the current and voltage for the series and parallel cells, you can just modify the parameters `Number of electrodes connected in parallel to make a cell` and `Number of cells connected in series to make a battery`. Note that this will simply calculate the current and voltage for the single cell you are modelling, so at the end you are simulating a single cell. For non-homogeneous behaviour you should use liionpack.

**Comment by Jadhav-Sourabh** [comment]
In liionpack, I am not able to perform CCCV experiment.
Please help me with this

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by balaji3112** [reply]
Did you find a solution? I'm currently working on a similar issue.

**Comment by Jadhav-Sourabh** [comment]
Yes

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by balaji3112** [comment]
Could you walk me through the steps that you took?

**Comment by balaji3112** [comment]
I would like to scale up my Thevenin cell model to become a battery pack. Since liionpack lacks a CCCV operating mode, I'm searching for a workaround using just pybamm.
